OverlayPlugin.Core/NetworkProcessors/LineMapEffect.cs

```diff
@@ -9,26 +11,449 @@ class LineMapEffect : LineBaseCustom<
+                case GameRegion.Chinese:
+                    {
+                        if (!helper.cn.ToStructs(message, out var header, out var packet))
+                            return false;
+                        WriteLinesFor(epoch, packet.instanceContentID, packet.count, packet.flags1, packet.flags2, packet.indexes);
+                        return true;
+                    }
+                case GameRegion.Korean:
+                    {
+                        if (!helper.kr.ToStructs(message, out var header, out var packet))
+                            return false;
+                        WriteLinesFor(epoch, packet.instanceContentID, packet.count, packet.flags1, packet.flags2, packet.indexes);
+                        return true;
+                    }
+
+                default:
+                    return false;
+            }
+
+            var line = packetHelper[currentRegion.Value].ToString(epoch, message);
+
+            if (line != null)
+            {
+                DateTime serverTime = ffxiv.EpochToDateTime(epoch);
+                logWriter(line, serverTime);
+                return true;
+            }
+            return false;
+        }
+
+        private void WriteLinesFor(long epoch, uint instanceContentID, ushort count, List<ushort> flags1, List<ushort> flags2, List<byte> indexes)
+        {
+            if (count == 0)
+            {
+                logger.Log(LogLevel.Error, $"Got MapEffect packet with 0 entries");
+                return;
+            }
+
+            if (flags1.Count != count)
+            {
+                logger.Log(LogLevel.Error, $"Mismatch in flags1 count for MapEffect packet, expected {count}, got {flags1.Count}");
+                return;
+            }
+            if (flags2.Count != count)
+            {
+                logger.Log(LogLevel.Error, $"Mismatch in flags2 count for MapEffect packet, expected {count}, got {flags2.Count}");
+                return;
+            }
+            if (indexes.Count != count)
+            {
+                logger.Log(LogLevel.Error, $"Mismatch in indexes count for MapEffect packet, expected {count}, got {indexes.Count}");
+                return;
+            }
+
+            DateTime serverTime = ffxiv.EpochToDateTime(epoch);
+
+            var strInstanceContentID = instanceContentID == 0 ? "" : $"{instanceContentID:X8}";
+
+            for (var i = 0; i < count; ++i)
+            {
+                var flag1 = flags1[i];
+                var flag2 = flags2[i];
+                var index = indexes[i];
+
+                // Two blank fields at the end used to be $"{unknown1:X2}|{unknown2:X4}", turned out to be padding
+                // @TODO: Remove these fields at some point, this is a breaking change.
+                // @TODO: Since we'll be breaking things anyways, maybe separate flags out into two fields?
+                logWriter($"{strInstanceContentID}|{flag1:X4}{flag2:X4}|{index:X2}||", serverTime);
```

Comment:
If bumping the logline version number, would it just make sense to do this now?  A quick search in the cactbot repo, and it looks like it wouldn't be a massive undertaking, probably could knock most or all of it out tonight and coordinate version bumps.  Don't know if/how many other plugins are using it though.

Reply:
I would rather not make a breaking change without giving more notice about it.